### PR TITLE
add support to generate docs from Cuttlefish

### PR DESCRIPTION
**Pull request by bookshelfdave**
This requires a PR from riak_ee, which I'll reference in a moment.

### CODE REVIEW

**Comment by bookshelfdave** [comment]
https://github.com/basho/riak_ee/pull/222

File: src/cuttlefish_rebar_plugin.erl [hunk 1]
```diff
@@ -71,6 +85,54 @@ generate(Config0, ReltoolFile) ->
     end,
     ok.
 
+render_templates(Config, Mappings) ->
+    PLs = lists:foldl(fun(M, Acc) ->
```
Comment (seancribbs) [inline src/cuttlefish_rebar_plugin.erl:89]: Is this not just a map/list-comp? Why is fold needed?

File: src/cuttlefish_rebar_plugin.erl [hunk 2]
```diff
@@ -71,6 +85,54 @@ generate(Config0, ReltoolFile) ->
...
+                       cuttlefish_conf:pretty_doc(cuttlefish_mapping:doc(M))},
+                      {see,
+                       cuttlefish_mapping:see(M)},
+                      {include_default,
+                       cuttlefish_mapping:include_default(M)}
+                     ] | Acc ]
+            end, [], Mappings),
+    CompileVars = [{mappings, PLs}],
+    Templates = rebar_config:get(Config, cuttlefish_doc_templates, []),
+    lists:foldl(fun({TemplateFile, OutputFile}, FileId) ->
```
Comment (seancribbs) [inline src/cuttlefish_rebar_plugin.erl:117]: What is the accumulator argument used for here? `FileId` seems to come in as `0` but becomes `ok` after the first iteration.

File: src/cuttlefish_rebar_plugin.erl (hunk 1 above)
Comment (bookshelfdave) [inline src/cuttlefish_rebar_plugin.erl:89]: fold isn't needed anymore, I'll change to a list-comp

File: src/cuttlefish_rebar_plugin.erl (hunk 2 above)
Comment (bookshelfdave) [inline src/cuttlefish_rebar_plugin.erl:117]: To guarantee ModName is unique. The fact that the acc is `ok` is a bug. It should increment with every template.

**Comment by bookshelfdave** [comment]
Rebased, submitting a new PR.

**Pull request closed without merging**